src/main/java/hudson/remoting/Engine.java

```diff
@@ -648,9 +648,18 @@ class Transport extends AbstractByteBufferCommandTransport {
                         }
                         @Override
                         protected void write(ByteBuffer header, ByteBuffer data) throws IOException {
-                            LOGGER.finest(() -> "sending message of length + " + ChunkHeader.length(ChunkHeader.peek(header)));
-                            session.getBasicRemote().sendBinary(header, false);
-                            session.getBasicRemote().sendBinary(data, true);
+                            LOGGER.finest(() -> "sending message of length " + ChunkHeader.length(ChunkHeader.peek(header)));
+                            // RemoteEndpoint.Async seems to lack isLast overloads
+                            // adapted from https://stackoverflow.com/a/42170003/12916
+                            ByteBuffer headerAndData = ByteBuffer.allocate(header.remaining() + data.remaining());
```

Comment:
This is of course inefficient—extra heap allocations.

I am trying to understand why https://github.com/jenkinsci/remoting/blob/2d296a8aa013fdceb1e3f6416a8006010dc83b8d/src/main/java/hudson/remoting/AbstractByteBufferCommandTransport.java#L296-L303 was written to use two `ByteBuffer`s when it looks like just one should suffice. Nothing in #92 makes it obvious.